src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9279,7 +9279,7 @@ bool Unit::HandleProcTriggerSpell(Unit* victim, uint32 damage, AuraEffect* trigg
             case SPELLFAMILY_SHAMAN:
                 {
                     // Lightning Shield (overwrite non existing triggered spell call in spell.dbc
-                    if (auraSpellInfo->SpellFamilyFlags[0] & 0x400)
+                    if (auraSpellInfo->SpellFamilyFlags[0] & 0x400 && auraSpellInfo->AttributesEx & 0x400)
```

Comment:
Do not use the hex code for Attributes, instead use the defined enum for it.

Defined in SharedDefines.h

Also you can use HasAttribute()